src/VirtualClient/VirtualClient.Actions/Blender/BlenderMetricsParser.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using VirtualClient.Actions.Blender;
+using VirtualClient.Contracts;
+
+namespace VirtualClient.Actions
+{
+    /// <summary>
+    /// Parser for the Blender workload.
+    /// </summary>
+    public class BlenderMetricsParser : MetricsParser
+    {
+        /// <summary>
+        /// Parser for the Blender workload
+        /// </summary>
+        /// <param name="rawText">The raw text from the Blender benchmark.</param>
+        public BlenderMetricsParser(string rawText)
+            : base(rawText)
+        {
+        }
+
+        /// <summary>
+        /// Parses the metrics from the workload results.
+        /// </summary>
+        /// <returns>List of parsed Metrics.</returns>
+        public override IList<Metric> Parse()
+        {
+            try
+            {
+                var metrics = new List<Metric>();
+                var blenderResults = JsonConvert.DeserializeObject<List<BlenderResult>>(this.RawText);
+                foreach (BlenderResult blenderResult in blenderResults)
+                {
+                    var metadata = new Dictionary<string, IConvertible>
+                    {
+                        { "blenderVersion", blenderResult.BlenderVersion.Version },
+                        { "benchmarkLauncher", blenderResult.BenchmarkLauncher.Label },
+                        { "scene", blenderResult.Scene.Label },
+                        // blender can only execute on one device(CPU/GPU) at a time
+                        { "deviceName", blenderResult.DeviceInfo.ComputeDevices[0].Name },
```

Comment:
Some of these might break if you don't have a GPU? Or do you parse all GPU if you have multiple ones?

Reply:
On a device without GPU, we will see results similar to:

`"device_info": {"device_type": "CPU", "compute_devices": [{"name": "AMD EPYC 7763 64-Core Processor", "type": "CPU"}], "num_cpu_threads": 32}}`

Which gives us the CPU score. I will add another example metrics file with CPU scores.